telemetry_device/StatisticsAnalyzer.cs

```diff
@@ -0,0 +1,63 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using telemetry_device.compactCollection;
+using telemetry_device_main.icds;
+
+namespace telemetry_device
+{
+    class StatisticsAnalyzer
+    {
+        private static StatisticsAnalyzer _instance;
+        private Dictionary<GlobalStatisticType,Statistic> _globalStatistics;
+        private Dictionary<IcdStatisticType,MultiStatistic> _icdStatistics;
+        private StatisticsAnalyzer()
+        {
+            _globalStatistics = new Dictionary<GlobalStatisticType, Statistic>();
+            _icdStatistics = new Dictionary<IcdStatisticType, MultiStatistic>();
+            foreach(GlobalStatisticType statType in Enum.GetValues(typeof(GlobalStatisticType)))
+                _globalStatistics.Add(statType, new Statistic());
+
+            foreach (IcdStatisticType statType in Enum.GetValues(typeof(IcdStatisticType)))
+                _icdStatistics.Add(statType, new MultiStatistic());
+        }
+        public static StatisticsAnalyzer Instance
+        {
+            get
+            {
+                if (_instance == null)
+                {
+                    _instance = new StatisticsAnalyzer();
+                }
+                return _instance;
+            }
+        }
+        public void UpdateStatistic(GlobalStatisticType statType,int value)
+        {
+            _globalStatistics[statType].AddCounter(value);
+        }
+        public void UpdateStatistic(IcdStatisticType statType,IcdTypes icdType,int value)
+        {
+            _icdStatistics[statType].AddCounter(icdType, value);
+        }
+        public Dictionary<string,float> GetDataDictionary()
+        {
+            Dictionary<string, float> avgDict = new Dictionary<string, float>();
```

Comment:
Not good the key of the dictionary will be string, put the key as the type of the object